runtime/queries/vue/highlights.scm

```diff
@@ -7,9 +7,25 @@
 (attribute
   (attribute_name) @attribute
   (quoted_attribute_value
-    (attribute_value) @string)
+    (attribute_value) @string)?
 )
 
+ (attribute
+  (attribute_name) @attribute
+)
+
+ (attribute
+   (attribute_name) @attribute
+   "=" @attribute_name
+   (#eq? @attribute_name "=")
+) @attribute
```

Comment:
Same here: this pattern is covered above. But capturing the entire `(attribute ...) @attribute` node means that everything in the node will be highlighted as an attribute, for example in `<style lang="css" scoped>` "lang" and "scoped" are highlighted (which we want) but so are the double quotes and "=" (which we should avoid).

Instead to match how we highlight HTML let's add a pattern at the bottom next to the `@punctuation.bracket` highlights:

```
"=" @punctuation.delimiter
```

so that the "=" has a highlight and modify the pattern on line 7 so that the `@string` highlight applies to the double quotes:

```
(attribute
  (attribute_name) @attribute
  [(attribute_value) (quoted_attribute_value)]? @string)
```

In that pattern I also changed the `(quoted_attribute_value)` part to be either `(quoted_attribute_value)` or `(attribute_value)` so that the highlight applies to `<style lang="css">` (quoted) and `<style lang=css>` (unquoted).

Reply:
Thank you for such a thorough review and tips on how to improve it!
Regarding the:
> but so are the double quotes and "=" (which we should avoid).

I was just going off the example how it is implemented in Neovim and VSCode (pic related), but I do agree that we should better avoid it.
![example](https://github.com/user-attachments/assets/a99d94c5-59b1-45f4-a6c3-33db2213b1f6)

Also, I do like how in helix - we have syntax highlighting for **directives**, here is how it looks in Neovim.
![example2](https://github.com/user-attachments/assets/5fd7fb32-6603-4f6f-97db-c2ce0985f84f)

And finally, helix uses grammar based on [ikatyang's](https://github.com/ikatyang/tree-sitter-vue/) implementation, which has not been updated over 3 years, nor issues were responded to. I feel like it might be outdated, but not sure

Reply:
If there's a competing Vue grammar that fixes the bugs of the one we use now we could consider switching to it. That can be done in a separate PR though